android/app/src/main/java/com/example/secretgenerator/MainActivity.kt

```diff
@@ -0,0 +1,70 @@
+package com.example.secretgenerator
+
+import android.app.Activity
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.widget.Button
+import android.widget.TextView
+import android.widget.Toast
+
+fun generateSecretKey(length: Int): String {
+    val charset = ('a'..'z') + ('A'..'Z') + ('0'..'9')
+
+    return List(length) { charset.random() }
+        .joinToString("")
+}
+
+class MainActivity : AppCompatActivity() {
+    private lateinit var keyTextView: TextView
+    private lateinit var errorGenerateTextView: TextView
+
+    private companion object {
+        const val REQUEST_SECRET_KEY = 0
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        keyTextView = findViewById(R.id.keyTextView)
+        errorGenerateTextView = findViewById(R.id.errorGenerateTextView)
```

Comment:
I don't like it. As far as I remember a better way to get views exists

Reply:
Maybe you talking about **View Binding?**

Reply:
guess so
maybe not
anyway you can resolve the thread if you think it is good way to get a view

Reply:
**View Binding** is a great way to get a view. But this task doesn't require View Binding usage and we better use a primitive way to get a view. But Kamil gave us a new task where we should use View Binding.
So this conversation can be closed.